notebooks/2.modeling/nbconvert/modeling.py

```diff
@@ -0,0 +1,435 @@
+    """Develops a Logistic Regression model employing the One vs Rest training
+    scheme and incorporates RandomizedSearchCV for optimal parameter selection.
+    This approach utilizes RandomizedSearchCV to explore a range of parameters
+    specified in the param_grid, ultimately identifying the most suitable model
+    configuration
+
+    This function will return best model.
+
+    Parameters
+    ----------
+    X_train : np.ndarray
+        Training features
+    y_train : np.ndarray
+        Training labels
+    param_grid : dict
+        parameters to tune
+    seed : Optional[int]
+        set random seed, default = 0
+
+    Returns
+    -------
+    BaseEstimator
+        Best model
+    """
+    # setting seed:
+    np.random.seed(seed)
+
+    # create a Logistic regression model with One vs Rest scheme (ovr)
+    logistic_regression_model = LogisticRegression(class_weight="balanced")
+    ovr_model = OneVsRestClassifier(logistic_regression_model)
+
+    # next is to use RandomizedSearchCV for hyper parameter turning
+    with parallel_backend("multiprocessing"):
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore", category=ConvergenceWarning, module="sklearn"
+            )
+            warnings.filterwarnings("ignore", category=UserWarning, module="sklearn")
+
+            # execute RandomizedResearchCV
+            random_search = RandomizedSearchCV(
+                estimator=ovr_model,
+                param_distributions=param_grid,
+                n_iter=10,
+                cv=5,
+                random_state=seed,
+                n_jobs=-1,
+            )
+
+            # fit with training data
+            random_search.fit(X_train, y_train)
+
+    # get the best model
+    best_model = random_search.best_estimator_
+    return best_model
+
+
+# In[3]:
+
+
+# setting paths and parameters
+results_dir = pathlib.Path("../../results").resolve(strict=True)
+data_splits_dir = (results_dir / "1.data_splits").resolve(strict=True)
+
+# setting path for training dataset
+training_dataset_path = (data_splits_dir / "training_data.csv.gz").resolve(strict=True)
+
+# holdout paths
+plate_holdout_path = (data_splits_dir / "plate_holdout.csv.gz").resolve(strict=True)
+treatment_holdout_path = (data_splits_dir / "treatment_holdout.csv.gz").resolve(
+    strict=True
+)
+wells_holdout_path = (data_splits_dir / "wells_holdout.csv.gz").resolve(strict=True)
+
+# setting output paths
+modeling_dir = (results_dir / "2.modeling").resolve()
+modeling_dir.mkdir(exist_ok=True)
+
+# ml parameters to hyperparameterization tuning
+param_grid = {
+    "estimator__C": uniform(0.1, 10),
+    "estimator__solver": ["newton-cg", "liblinear", "sag", "saga"],
+    "estimator__penalty": ["l1", "l2", "elasticnet"],
+    "estimator__l1_ratio": uniform(0, 1),
+}
+
+
+# In[4]:
+
+
+# loading injury codes
+with open(data_splits_dir / "injury_codes.json") as json_file:
+    injury_codes = json.load(json_file)
```

Comment:
Would it make sense to use [`pd.read_json()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html) here instead of the call to `open`?